packages/wonder-blocks-popover/components/popover-content.js

```diff
@@ -0,0 +1,193 @@
+
+type CommonProps = {|
+    ...AriaProps,
+
+    /**
+     * The content to render inside the popover.
+     */
+    content: string,
+
+    /**
+     * The popover title
+     */
+    title: string,
+
+    /**
+     * User-defined actions.
+     */
+    actions?: React.Node,
+
+    /**
+     * Close button label for use in screen readers
+     */
+    closeButtonLabel?: string,
+
+    /**
+     * When true, the close button is shown; otherwise, the close button is not shown.
+     */
+    closeButtonVisible?: boolean,
+
+    /**
+     * Called when the popover closes
+     */
+    onClose?: () => mixed,
+
+    /**
+     * Custom styles
+     */
+    style?: StyleType,
+
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string,
+
+    /**
+     * Without these, flow complains about icon, image and emphasized not being
+     * available on props at all b/c these are exact object types.
+     */
+    icon?: void,
+    image?: void,
+    emphasized?: void,
+|};
+
+type WithEmphasized = {|
+    ...CommonProps,
+
+    /**
+     * When true, changes the popover window background to blue; otherwise, the
+     * popover window background is not modified. It can be used only with
+     * Text-only popovers. It cannot be used with icon or image.
+     */
+    emphasized: boolean,
+|};
+
+type WithIcon = {|
+    ...CommonProps,
+
+    /**
+     * Decorate the popover with an illustrated icon. It cannot be used at the
+     * same time with image.
+     */
+    icon?: string,
+|};
+
+type WithImage = {|
+    ...CommonProps,
+
+    /**
+     * Decorate the popover with a full-bleed illustration. It cannot be used at
+     * the same time with icon.
+     */
+    image?: string,
+|};
+
+type Props = CommonProps | WithEmphasized | WithIcon | WithImage;
+
+// Created to add custom styles to the icon or image elements
+const StyledImage = addStyle("img");
+
+/**
+ * This is the container that is consumed by all the predefined variations. Its
+ * main responsibility is populate the contents depending on the variation used.
+ */
+export default class PopoverContent extends React.Component<Props> {
+    static defaultProps = {
+        closeButtonVisible: false,
+    };
+
+    render() {
+        const {
+            actions,
+            closeButtonLabel,
+            closeButtonVisible,
+            content,
+            emphasized,
+            icon,
+            image,
+            onClose,
+            style,
+            title,
+            testId,
+        } = this.props;
+
+        return (
+            <PopoverContentCore
+                color={emphasized ? "blue" : "light"}
+                closeButtonLabel={closeButtonLabel}
+                closeButtonVisible={closeButtonVisible}
+                onClose={onClose}
+                style={style}
+                testId={testId}
+            >
+                <View style={!!icon && styles.withIcon}>
+                    {image && <StyledImage style={styles.image} src={image} />}
```

Comment:
**Known issue:** When adding an image, the close button  color is dark. It should be white for this scenario.
![popover-content-illustration-issue](https://user-images.githubusercontent.com/843075/59958473-6a7a5f80-9475-11e9-94fc-3bb36ea8206d.png)


I expect to fix this as part of PR 3/4. Need to figure out a way to pass a closeButton color from the `PopoverContent` to the `PopoverContentCore` component.